tests/pipeline/test_optimize.py

```diff
@@ -138,3 +140,36 @@ def test_no_sklearn_head_does_not_trigger_threadpoolctl(tmp_path: Path) -> None:
 
     report = history[0]
     assert report.summary["num_threads"] == num_threads
+
+
+def test_optimizer_is_reported_to(tmp_path: Path) -> None:
+    class MyOptimizer(SMACOptimizer):
+        def __init__(
+            self,
+            *args: Any,
+            **kwargs: Any,
+        ) -> None:
+            self.told_report: Trial.Report | None = None
+            super().__init__(*args, **kwargs)
+
+        @override
+        def tell(self, report: Trial.Report) -> None:
+            self.told_report = report
+            return super().tell(report)
```

Comment:
Custom optimizer we can inspect post fact, it just records the report it was `tell`'ed about, defaulting to SMAC optimizer for everything else.